File: dotnet/DesignPatterns/StructuralDesignPatterns/FacadeDesignPattern/FacadeDesignPattern.Example1/Models/DvdPlayer.cs
namespace FacadeDesignPattern.Example1.Models;

public class DvdPlayer
{
	public void On() => Console.WriteLine("DVD player is turned on.");
	public void Off() => Console.WriteLine("DVD player is turned off.");
	public void Play(string movie) => Console.WriteLine($"Playing movie: '{movie}'.");
	public void Stop() => Console.WriteLine("Movie playback stopped.");
}

File: nodejs/DesignPatterns/StructuralDesignPatterns/FacadeDesignPattern/FacadeDesignPattern.Example1/src/models/DvdPlayer.ts
export class DvdPlayer {
  on(): void { console.log('DVD player is turned on.'); }
  off(): void { console.log('DVD player is turned off.'); }
  play(movie: string): void { console.log(`Playing movie: '${movie}'.`); }
  stop(): void { console.log('Movie playback stopped.'); }
}
